storage/buckets.rb

```diff
@@ -199,6 +199,133 @@ def delete_bucket project_id:, bucket_name:
   # [END delete_bucket]
 end
 
+def set_retention_policy project_id:, bucket_name:, retention_period:
+  # [START storage_set_retention_policy]
+  # project_id       = "Your Google Cloud project ID"
+  # bucket_name      = "Name of your Google Cloud Storage bucket"
+  # retention_period = "Object retention period defined in seconds"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+  bucket  = storage.bucket bucket_name
+
+  bucket.retention_period = retention_period
+
+  puts "Retention period for #{bucket_name} is now #{retention_period} seconds"
+  # [END storage_set_retention_policy]
+end
+
+def lock_retention_policy project_id:, bucket_name:
+  # [START storage_lock_retention_policy]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Name of your Google Cloud Storage bucket"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+  bucket  = storage.bucket bucket_name
+
+  # Warning: Once a retention policy is locked it cannot be unlocked
+  # and retention period can only be increased.
+  # Uses bucket.metageneration as a precondition
```

Comment:
This sentence is missing a period at the end.

Reply:
If you mean the public Ruby API, bucket.metageneration should be `Bucket#metageneration`.